internal/controller/repository/repository.go

```diff
@@ -361,6 +457,69 @@ func updateRepoTeams(ctx context.Context, cr *v1alpha1.Repository, gh *ghclient.
 	return nil
 }
 
+func updateRepoWebhooks(ctx context.Context, cr *v1alpha1.Repository, gh *ghclient.Client, repoName string) error {
+	crWToConfig := getWebhookMapFromCr(cr.Spec.ForProvider.Webhooks)
+	ghWToConfig, err := getRepoWebhooksWithConfig(ctx, gh, cr.Spec.ForProvider.Org, repoName)
+	if err != nil {
+		return err
+	}
+
+	toDelete, toAdd, toUpdate := util.DiffRepoWebhooks(ghWToConfig, crWToConfig)
+
+	for name, _ := range toDelete {
+		id, err := getRepoWebhookId(ctx, gh, cr.Spec.ForProvider.Org, repoName, name)
+		if err != nil {
+			return err
+		}
+		if id == nil {
```

Comment:
`id == nil` should be an error condition

Reply:
Fixed